src/main/java/xyz/aadev/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -174,29 +174,69 @@ public boolean isItemValidForSlot(int slot, ItemStack itemStack) {
         if (side == EnumFacing.UP && machineTier == MachineTier.TIER_0) {
             slots = new int[1];
         }
-        if (side == EnumFacing.DOWN) {
-            slots = new int[2];
-            slots[0] = 2;
-            slots[1] = 3;
+        int i = 0;
+        for (final EnumFacing sidea : EnumFacing.VALUES) {
+            if (sidea == side && (sides[i]==1 || sides[i]==0)) {
+                slots = new int[2];
+                slots[0] = 2;
+                slots[1] = 3;
+            }
+            i++;
         }
 
         return slots;
     }
 
+
     @Override
     public Object getClientGuiElement(int guiId, EntityPlayer player) {
-        return new GuiPulverizer(player.inventory, this);
+        if (guiId==0){
+            return new GuiPulverizer(player.inventory, this);
+        }
+        if (guiId==3){
+            return new GuiPowerStorage(player.inventory, this,sides,5,player);
+        }
+        return null;
     }
 
     @Override
     public Object getServerGuiElement(int guiId, EntityPlayer player) {
-        return new ContainerPulverizer(player.inventory, this);
+        if (guiId==0){
+            return new ContainerPulverizer(player.inventory, this);
+        }
+        if (guiId==3){
+            return new ContanierPowerStorage(player.inventory,this,5);
+        }
+        return null;
     }
 
     @Override
     public boolean canExtractItem(int index, ItemStack stack, EnumFacing direction) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "canExtractItem" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)